### DISCUSSION TITLE
Is it a good idea to run hydra and kratos on the same Postgresql database?

### DISCUSSION THREAD

**Discussion by CMCDragonkai** [ory/hydra#2558] [Q&A] [answered]
I am planning to run both Kratos and Hydra at the same time, and since the authentication system needs to  do both activities I thought I could run on the same database.

I noticed that I could run both migrations against the same postgresql database, however I'm unsure if this is a good idea.

I noticed that when I'm using hydra migrate the tables are all prefixed with `hydra_` which appears to prevent conflicts, however I do see these SQL errors:

```
2021-06-03 06:43:39.595 UTC [11204] ERROR:  relation "hydra_client_migration" does not exist at character 15
2021-06-03 06:43:39.595 UTC [11204] STATEMENT:  SELECT * FROM hydra_client_migration
The following migration is planned:
...
2021-06-03 06:43:39.602 UTC [11204] ERROR:  relation "hydra_client_migration" does not exist at character 15
2021-06-03 06:43:39.602 UTC [11204] STATEMENT:  SELECT * FROM hydra_client_migration
Successfully applied migrations!
```

**Comment by aeneasr** [comment, marked as answer]
You can certainly run them on a single database instance (so e.g. one docker container) but it's not recommended to run them against the same database schema. It is better to do `CREATE DATABASE <service-name>` to keep isolation and potential conflicts out.

**Reply by CMCDragonkai** [reply]
I was started doing that since I noticed both schemas have migrations.

However do you have any information about those error messages?

**Reply by CMCDragonkai** [reply]
These ones: relation `"hydra_client_migration" does not exist at character 15`
